Sources/edge/Commands/Devices.swift

```diff
@@ -0,0 +1,152 @@
+import ArgumentParser
+import Foundation
+import Logging
+
+protocol DeviceDiscovery {
+    func findUSBDevices(logger: Logger) async -> [USBDevice]
+    func findEthernetInterfaces(logger: Logger) async -> [EthernetInterface]
+}
+
+struct DevicesCommand: AsyncParsableCommand {
+    static let configuration = CommandConfiguration(
+        commandName: "devices",
+        abstract: "List USB and Ethernet devices connected to the system"
+    )
+
+    enum DeviceType: String, ExpressibleByArgument {
+        case usb, ethernet, all
+    }
+
+    enum OutputFormat: String, ExpressibleByArgument {
+        case json, text
+    }
+
+    @Option(help: "Device types to list (usb, ethernet, or both)")
+    var type: DeviceType = .all
+
+    @Flag(name: [.customShort("j"), .long], help: "Output in JSON format")
+    var json: Bool = false
+
+    func listDevices(
+        usbDevices: [USBDevice],
+        ethernetInterfaces: [EthernetInterface],
+        logger: Logger
+    ) async {
+        let format = json ? OutputFormat.json : OutputFormat.text
+
+        // If using JSON format and we have both types of devices,
+        // use the DevicesCollection struct for combined output
+        if format == .json && type == .all {
+            let collection = DevicesCollection(usb: usbDevices, ethernet: ethernetInterfaces)
+            do {
+                let jsonString = try collection.toJSON()
+                print(jsonString)
+            } catch {
+                logger.error("Error serializing to JSON: \(error)")
+            }
+            return
+        }
+
+        // Otherwise, handle each type separately
+        if type == .usb || type == .all {
+            await listUSBDevices(devices: usbDevices, logger: logger)
+        }
+
+        if type == .ethernet || type == .all {
+            await listEthernetInterfaces(interfaces: ethernetInterfaces, logger: logger)
+        }
+    }
+
+    func listUSBDevices(devices: [USBDevice], logger: Logger) async {
+        if devices.isEmpty {
+            let message = "No EdgeOS devices found."
+            logger.info("\(message)")
+            print(message)
+            return
+        }
```

Comment:
This isn't true, we might not have found USB devices but still find devices on the LAN